### ISSUE TITLE
Parameters of node2vec

### ISSUE THREAD

**Issue by MasterManggis** [xuanyuan14/CACM-master#6]
In order to better reproduce your wonderful work, could you provide the training parameters of node2vec, the known parameters are -d:64 -dr -w, how do you set other such as -l,-r,-k,-e,-p,-q?

**Comment by xuanyuan14** [comment]
Hi~Thanks for your attention! We just use the default setting of other parameters.

**Comment by MasterManggis** [comment]
Thank you for your quick answer.
Due to the slow training speed when running the model (I use TITAN RTX graphics card), I read the code by the way and found two confusing problems:
1. You mentioned in the paper that there are 19 vertical types with dim=8, but in fact, there are 5174 vertical types in the vtype_vid.dict file. Vertical types are represented and mapped like `<vtype>A#B</vtype>` in the dataset. I am curious about the source of 19 vertical types.
2. In the paper, click perplexity (PPL) at rank r is calculated with the following formula
![A1](https://user-images.githubusercontent.com/56973432/147447388-97dff8b2-d91c-4353-86f0-bdc57f1f2a9f.png)
    The exponential is the opposite of the average entropy of the same rank and different query&session. But in the code it is calculated as follow(model.py line101)
    
    `perplexity_at_rank[position_idx % 10] += torch.log2(pred_scores[0][position_idx % 10].view(1) + MINF)` 
    
    I personally think that the %10 on the right should be deleted, because pred_scores[0] is a 1d vector and all doc of current session, adding %10 means that the click rate of doc with an index greater than 10 will never be calculated, and other mistakes will be caused.

**Comment by xuanyuan14** [comment]
1. 19 is a general number of the category of vertical results. However, 5174 is a more specific number, which means many types within 5174 can be generalized into one of the 19 types.
2. The position_idx will be greater than 10 if there are more than one query within the session. However, we only aim at calculating PPL@r where r <= 10, i.e., the perplexity with one query. Therefore, we need to ground all session position_idxes via mod 10.
My WeChat account is xuanyuan1414. If you have any problem, please feel free to contact me.
